### ISSUE TITLE
The per-task activation backstop is prose in the same file as the rule it guards, so it fails together with it: a Stop hook checking whether the log was touched would be a second layer on a harness event

### ISSUE THREAD

**Issue by psycoban** [rebelytics/one-skill-to-rule-them-all#192]
**Adjacent to #62, #177 and #185, distinct from all three.** Those are about activation never happening or an activation tier breaking. This one is about the layer that is supposed to *discover* a missed activation.

## What happened

A long interactive Claude Code session (sync work, a multi-agent review, a fix, a pull request, an analysis: seven task boundaries, well over a hundred tool calls) never invoked the skill. Both tiers were configured and both fired: the instruction-file block ("before the first tool call ... invoke the skill AND execute its Session Start Protocol") was in context, and the session-start hook injected its reminder. Neither was acted on.

The per-task backstop in the same instruction block did not reveal the miss:

> After completing each task, check the observation records written this session and report a one-line summary ... This is the activation backstop: it forces a look at the log, so a session that silently skipped the protocol is discovered at the first task boundary instead of never.

It was skipped at all seven boundaries. Activation only happened when the model was switched mid-session and the new model read the same instructions.

## Why the current text did not prevent it

The backstop is written in the same medium as the rule it guards. It is prose in the instruction file, so it depends on exactly the behaviour that had already failed: the agent re-reading and acting on that block. Whatever made the session skip the first paragraph made it skip the second. The session-start hook is a harness event, but it fires once, before any work exists, and its output is again prose the agent may not act on.

`references/environments.md` already says the harness hook is "the only enforced option" for activation. The same reasoning applies to the backstop, and nothing in the references offers a harness-event form of it.

## Suggested fix

Offer a Stop-hook recipe next to the prose backstop in `references/environments.md`. The check it needs is cheap and objective: *has this session used tools, and has anything under the observation log or the checkpoint file been written since the session started?*

This is the hook I am now running (Node, no dependencies; generalised paths). It blocks the stop once per session with a one-line instruction, and stays silent for conversational turns, for sessions that already wrote to the log, and when the log is unreachable:

```js
// Stop hook: the per-task backstop as a harness event instead of prose.
const fs = require('node:fs'), os = require('node:os'), path = require('node:path');
const MIN_TOOL_USES = 3; // below this the turn is conversation, not work

let input = {};
try { input = JSON.parse(fs.readFileSync(0, 'utf8') || '{}'); } catch { process.exit(0); }
if (input.stop_hook_active) process.exit(0);            // never chain a second block
const session = String(input.session_id || '').replace(/[^A-Za-z0-9_-]/g, '');
const transcript = input.transcript_path;
if (!session || !transcript) process.exit(0);

const marker = path.join(os.tmpdir(), `task-observer-backstop-${session}`);
if (fs.existsSync(marker)) process.exit(0);             // once per session

let started, toolUses;
try {
  started = fs.statSync(transcript).birthtimeMs;
  toolUses = (fs.readFileSync(transcript, 'utf8').match(/"type":\s*"tool_use"/g) || []).length;
} catch { process.exit(0); }
if (toolUses < MIN_TOOL_USES) process.exit(0);

const base = path.join(os.homedir(), '.claude', 'skill-observations'); // the pinned workspace
const files = [path.join(base, 'checkpoints.log')];
try {
  for (const n of fs.readdirSync(path.join(base, 'observation-log')))
    if (n.endsWith('.md')) files.push(path.join(base, 'observation-log', n));
} catch { process.exit(0); }                             // unreachable log: stay silent
const touched = files.some((f) => { try { return fs.statSync(f).mtimeMs > started; } catch { return false; } });
if (touched) process.exit(0);

fs.writeFileSync(marker, new Date().toISOString());
console.log(JSON.stringify({ decision: 'block', reason:
  'task-observer backstop: this session used tools but wrote no observation file and no checkpoint line. ' +
  'If the skill was never invoked, invoke it now; then write pending observations or append a ' +
  '"no observations" line to checkpoints.log, and report the one-line summary. Fires once per session.' }));
```

Controls I ran before registering it, since a trigger is only armed once it has matched the real event (your own rule in "How to Log"):

| case | expected | result |
|---|---|---|
| 5 tool uses, nothing written since the transcript was created | block | block |
| same session again | silent (marker) | silent |
| 1 tool use | silent | silent |
| `stop_hook_active: true` | silent | silent |
| checkpoint file touched after session start | silent | silent |
| the literal pattern against a real transcript | > 0 matches | 230 matches |

Known ceiling, stated in the script I keep locally: it uses mtime, so a parallel session writing to the same log counts as "touched". Keying the check on the session id inside `checkpoints.log` would remove that false negative.

## Principle

A backstop written in the same medium as the rule it guards fails together with it. The second layer has to hang on an event the harness raises, not on the agent re-reading its instructions. Classify each enforcement layer by what it needs in order to function: a filesystem write survives into any session type, a prose reminder survives only while the prose is being acted on.

**Comment by rebelytics** [comment]
Thanks, and thanks for shipping the hook alongside the report. In the queue for one of the next weekly releases.
